fastai/optimizer.py

```diff
@@ -11,12 +11,24 @@
 # Cell
 class _BaseOptimizer():
     "Common functionality between `Optimizer` and `OptimWrapper`"
-    def all_params(self, n=slice(None), with_grad=False):
+    def all_params(self,
+        n=slice(None), # Where to slice the model or all as default
```

Comment:
I don't think this comment is correct. It's not a "where to slice", the `slice` is like for indexing (`a[slice(None)]=a[:]`). So you're indexing through the params.

Reply:
mmmmm, @tmabraham  so it is not slicing the model, but the params of a specific layer?

Reply:
@tyoc213 Yeah I think it's just indexing specific layers. Maybe check for yourself if this is true? It's been a while since I played around with the fastai Optimizer system...

Reply:
Im changing this to

```
def all_params(self,
        n:slice=slice(None), # extended slicing over the optimizer `param_lists`
        with_grad:bool=False # Get all param tuples. If `True` select only those with a gradient
    ):
```